hypyp/prep.py

```diff
@@ -229,89 +244,90 @@ def AR_local(cleaned_epochs_ICA: list, strategy:str = 'union', threshold:float =
-        bad_epochs_AR.append(reject_log)
+        ar.fit(clean_epochs_subj)
+        reject_log = ar.get_reject_log(clean_epochs_subj, picks=picks)
 
-    # taking bad epochs for min 1 subj (dyad)
-    log1 = bad_epochs_AR[0]
-    log2 = bad_epochs_AR[1]
+        AR.append(ar)
+        reject_logs.append(reject_log)
 
-    bad1 = np.where(log1.bad_epochs == True)
-    bad2 = np.where(log2.bad_epochs == True)
+    # taking bad epochs for min 1 subj (dyad)
+    bad1_idx = np.where(reject_logs[0].bad_epochs == True)[0].tolist()
+    bad2_idx = np.where(reject_logs[1].bad_epochs == True)[0].tolist()
 
     if strategy == 'union':
-        bad = list(set(bad1[0].tolist()).union(set(bad2[0].tolist())))
+        bad_idx = list(set(bad1_idx).union(set(bad2_idx)))
     elif strategy == 'intersection':
-        bad = list(set(bad1[0].tolist()).intersection(set(bad2[0].tolist())))
+        bad_idx = list(set(bad1_idx).intersection(set(bad2_idx)))
     else:
-        TypeError('not good strategy input!')
+        raise RuntimeError('not good strategy input!')
 
     # storing the percentage of epochs rejection
-    dic_AR['S1'] = float((len(bad1[0].tolist())/len(cleaned_epochs_ICA[0]))*100)
-    dic_AR['S2'] = float((len(bad2[0].tolist())/len(cleaned_epochs_ICA[1]))*100)
+    dic_AR['S1'] = float((len(bad1_idx) / len(cleaned_epochs_ICA[0])) * 100)
+    dic_AR['S2'] = float((len(bad2_idx) / len(cleaned_epochs_ICA[1])) * 100)
 
     # picking good epochs for the two subj
-    cleaned_epochs_AR = []
-    for clean_epochs in cleaned_epochs_ICA:  # per subj
+    cleaned_epochs_AR: List[mne.Epochs] = []
+    for subject_id, clean_epochs_subj in enumerate(cleaned_epochs_ICA):  # per subj
         # keep a copy of the original data
-        clean_epochs_ep = copy.deepcopy(clean_epochs)
-        clean_epochs_ep = clean_epochs_ep.drop(indices=bad)
+        epochs_subj = copy.deepcopy(clean_epochs_subj)
+        epochs_subj.drop(indices=bad_idx)
         # interpolating bads or removing epochs
-        ar = AR[cleaned_epochs_ICA.index(clean_epochs)]
-        clean_epochs_AR = ar.transform(clean_epochs_ep)
-        cleaned_epochs_AR.append(clean_epochs_AR)
+        ar = AR[subject_id]
+        epochs_AR = ar.transform(epochs_subj)
+        cleaned_epochs_AR.append(epochs_AR)
 
     if strategy == 'intersection':
         # equalizing epochs length between two participants
         mne.epochs.equalize_epoch_counts(cleaned_epochs_AR)
 
-    dic_AR['dyad'] = float(((len(cleaned_epochs_ICA[0])-len(cleaned_epochs_AR[0]))/len(cleaned_epochs_ICA[0]))*100)
+    n_epochs_ICA = len(cleaned_epochs_ICA[0])
+    n_epochs_AR = len(cleaned_epochs_AR[0])
+
+    dic_AR['dyad'] = float(((n_epochs_ICA - n_epochs_AR) / n_epochs_ICA) * 100)
     if dic_AR['dyad'] >= threshold:
-        TypeError('percentage of rejected epochs above threshold!')
+        raise RuntimeError(f"percentage of rejected epochs ({dic_AR['dyad']}) above threshold ({threshold})! ")
```

Comment:
There was a bug here: Errors need to be raised to interrupt the execution flow. The test on this condition was inadequate (also fixed in this commit).

Also the type `RuntimeError` seems more appropriate.